File: TiaCodegen/Commands/Functions/PTrigCall.cs
using TiaCodegen.Commands.Coils;
using TiaCodegen.Commands.Signals;
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands.Functions
{
    public class PTrigCall : BaseNPCoil, IPartName
    {
        public PTrigCall(Signal signal) : base(signal, null, null)
        {
            PartName = "PBox";
        }

        public string PartName { get; set; }
    }
}

File: TiaCodegen-ts/src/Commands/Functions/PTrigCall.ts
import { Signal } from '../Signals/Signal.js';
import { IPartName } from '../../Interfaces/IPartName.js';
import { BaseNPCoil } from '../Coils/BaseNPCoil.js';

export interface PTrigCallOptions {
    signal: Signal;
}

export class PTrigCall extends BaseNPCoil implements IPartName {
    partName: string;

    constructor(options: PTrigCallOptions) {
        const { signal } = options;
        super(signal, null, null);
        this.partName = 'PBox';
    }
}
